main.go

```diff
@@ -18,143 +18,49 @@
 package main
 
 import (
-	"context"
-	"errors"
 	"flag"
 	"fmt"
-	"io/ioutil"
 	"log"
+	"net"
 	"os"
-	"os/signal"
-	"path/filepath"
-	"syscall"
+	"path"
 
-	secretmanager "cloud.google.com/go/secretmanager/apiv1"
-	"github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/auth"
-	"github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/config"
-	"golang.org/x/oauth2"
-	"google.golang.org/api/option"
+	"github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/server"
+
+	"google.golang.org/grpc"
+	"sigs.k8s.io/secrets-store-csi-driver/provider/v1alpha1"
 )
 
 var (
 	daemonset  = flag.Bool("daemonset", false, "Controls whether the plugin executes in the DaemonSet mode, copying itself to TARGET_DIR")
 	kubeconfig = flag.String("kubeconfig", "", "absolute path to kubeconfig file")
 
-	attributes = flag.String("attributes", "", "Secrets volume attributes.")
-	secrets    = flag.String("secrets", "", "Kubernetes secrets passed through the CSI driver node publish interface.")
-	targetPath = flag.String("targetPath", "", "Path to where the secrets should be written")
-	permission = flag.Uint("permission", 700, "File permissions of the written secrets")
-
 	version = "dev"
 )
 
-// The "provider" name in the "SecretProviderClass" CRD that this plugin
-// operates on.
-const providerName = "gcp"
-
 func main() {
 	flag.Parse()
-	// https://github.com/kubernetes-sigs/secrets-store-csi-driver only logs
-	// stderr if the plugin execution is not successful.
-	log.SetOutput(os.Stdout)
-	ctx := withShutdownSignal(context.Background())
 
 	ua := fmt.Sprintf("secrets-store-csi-driver-provider-gcp/%s", version)
 	log.Printf("starting %s", ua)
 
-	// This plugin and the github.com/kubernetes-sigs/secrets-store-csi-driver
-	// driver are both installed as DaemonSets that share a common folder on
-	// the host. When the "-daemonset" flag is set this binary copies itself
-	// to the TARGET_DIR folder and sleeps indefinitely.
-	if *daemonset {
-		if err := copyself(ctx); err != nil {
-			log.Fatal(err)
-		}
+	if !*daemonset {
```

Comment:
Is this for backwards compatibility? Should we just remove the flag or leave the declaration but ignore it and always run in daemonset mode?

Reply:
Ah pushed a change to remove that